pkg/machine/ignition/ignition.go

```diff
@@ -685,6 +685,56 @@ done
 `
 }
 
+func (i *IgnitionBuilder) AddPlaybook(input *os.File, destPath string, username string) error {
+	// read the config file to a string
+	s, err := os.ReadFile(input.Name())
+	if err != nil {
+		return err
+	}
```

Comment:
The caller opens the file then passes the file and then you disregard that and open it again.
Either the caller passes the name directly or use `io.ReadAll()` so we do not open the file twice